src/oadp_resources/oadp/csi/volume_snapshot_class.py

```diff
@@ -0,0 +1,46 @@
+from oadp_constants.resources import ApiGroups
+from ocp_resources.resource import NamespacedResource
+from enum import Enum
+
+
+class VolumeSnapshotClass(NamespacedResource):
+    api_group = ApiGroups.VOLUME_SNAPSHOT_CLASS.value
+
+    class Platform(Enum):
+        class AWS(Enum):
+            PROVIDER = "aws"
+            DRIVER = "ebs.csi.aws.com"
+
+        class GCP(Enum):
+            PROVIDER = "gcp"
+            DRIVER = "pd.csi.storage.gke.io"
+
+        class AZURE(Enum):
+            PROVIDER = "azure"
+            DRIVER = "disk.csi.azure.com"
+
+        class OSP(Enum):
+            PROVIDER = "osp"
+            DRIVER = "openshift-storage.rbd.csi.ceph.com"
+
+        class IBM(Enum):
+            PROVIDER = "osp"
+            DRIVER = "openshift-storage.cephfs.csi.ceph.com"
+
+    @classmethod
+    def get_csi_driver_by_provider(cls, provider):
```

Comment:
Mmmm
it is stattis
maybe the client the provier = GCP.DRIVER
in stead of having this nested it?